src/watertap_contrib/reflo/unit_models/tests/test_solar_still.py

```diff
@@ -0,0 +1,247 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import os
+import pytest
+
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    assert_optimal_termination,
+    units as pyunits,
+)
+from pyomo.network import Port
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock, UnitModelCostingBlock, MaterialFlowBasis
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util.exceptions import InitializationError, ConfigurationError
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+)
+
+from watertap.core.solvers import get_solver
+from watertap.property_models.multicomp_aq_sol_prop_pack import (
+    MCASParameterBlock,
+)
+
+from watertap_contrib.reflo.unit_models import SolarStill
+from watertap_contrib.reflo.costing import TreatmentCosting
+
+# Get default solver for testing
+solver = get_solver()
+
+__location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+test_data_path = f"{__location__}/solar_still_test_data_phoenix_az.csv"
+
+
+def build_ss():
+
+    rho = 1000 * pyunits.kg / pyunits.m**3
+    inlet_dict = {
+        "solute_list": ["TDS"],
+        "mw_data": {"TDS": 31.4038218e-3},
+        "material_flow_basis": MaterialFlowBasis.mass,
+    }
+
+    water_yield_calc_dict = dict(
+        input_weather_file_path=test_data_path,  # path to input weather file
+        interval_day=15,  # interval at which to calculate water yield (e.g., every 15 days)
+        salinity=20,  # salinity of influent water; g/L
```

Comment:
We could probably do away with this salinity key and use total_dissolved_solids, for example, under the hood of the unit model. My broader point is whether this can be more of an optional arg that would only be specified if the salinity input couldn't be obtained from the property model by the unit model. 

This is actually a good example where you might want to manually enter the salinity as you did here since TDS is the solute, and currently, MCAS may or may not output the right value for total_dissolved_solids. (Note that I am not considering complexities such as organics contributing to TDS and just thinking of salinity here).

Reply:
This has to be a manual entry since it goes into the helper function to calculate the water yield, and passing e.g., `conc_mass_phase_comp["Liq", "TDS"]` wouldn't work since the value as passed during `build` would be the initialized value of the variable rather than the actual value of TDS for the flowsheet. 

(but maybe I misunderstood your point)